zoltar-metrics/src/main/java/com/spotify/zoltar/metrics/semantic/SemanticPredictorMetrics.java

```diff
@@ -40,40 +45,73 @@
 @AutoValue
 public abstract class SemanticPredictorMetrics implements PredictorMetrics {
 
-  abstract Timer predictDurationTimer();
-
-  abstract Meter predictRateCounter();
-
-  abstract Timer extractDurationTimer();
-
-  abstract Meter extractRateCounter();
+  abstract LoadingCache<Id, Meters> metersCache();
 
   /** Creates a new @{link SemanticPredictorMetrics}. */
-  public static SemanticPredictorMetrics create(final SemanticMetricRegistry registry) {
-    final MetricId predictDurationId = MetricId.build().tagged("what", PREDICT_DURATION.tag());
-    final MetricId predictRateId = MetricId.build().tagged("what", PREDICT_RATE.tag());
-    final MetricId extractDurationId = MetricId.build()
-        .tagged("what", FEATURE_EXTRACT_DURATION.tag());
-    final MetricId extractRateId = MetricId.build().tagged("what", FEATURE_EXTRACT_RATE.tag());
-
-    final Timer predictDurationTimer = registry.timer(predictDurationId);
-    final Meter predictRateCounter = registry.meter(predictRateId);
-    final Timer extractDurationTimer = registry.timer(extractDurationId);
-    final Meter extractRateCounter = registry.meter(extractRateId);
-
-    return new AutoValue_SemanticPredictorMetrics(predictDurationTimer,
-                                                  predictRateCounter,
-                                                  extractDurationTimer,
-                                                  extractRateCounter);
+  public static SemanticPredictorMetrics create(final SemanticMetricRegistry registry,
+                                                final MetricId metricId) {
```

Comment:
Can we add a `create(final SemanticMetricRegistry registry)` that calls this one with `MetricId.EMPTY`?

Reply:
We can.

Reply:
Hm now that I think about it..does it make sense? If a bunch of different services create `SemanticMetricRegistry` their metrics would all get combined right? Maybe we could default to service name?

Reply:
if you run multiple services with the model default ids than yes they would be aggregated. Personally i would leave it as a param.

In this example, sure we can put in the service name.

Reply:
👍